circuit-shared-elements/src/commonMain/kotlin/com/slack/circuit/sharedelements/DelicateCircuitSharedElementsApi.kt

```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2024 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuit.sharedelements
+
+import kotlin.annotation.AnnotationTarget.FUNCTION
+import kotlin.annotation.AnnotationTarget.PROPERTY
+
+/** Indicates that the annotated shared elements API is delicate and should be used carefully. */
+@RequiresOptIn
+@Retention(AnnotationRetention.BINARY)
+@Target(FUNCTION, PROPERTY)
+public annotation class DelicateCircuitSharedElementsApi
```

Comment:
Let's go the whole way and make this just `ExperimentalCircuitSharedElementsApi` for now